centreon/src/Core/TimePeriod/Application/UseCase/AddTimePeriod/AddTimePeriodRequest.php

```diff
@@ -23,18 +23,66 @@
 
 namespace Core\TimePeriod\Application\UseCase\AddTimePeriod;
 
+use Symfony\Component\Validator\Constraints as Assert;
+
 final class AddTimePeriodRequest
 {
-    public string $name = '';
-
-    public string $alias = '';
-
-    /** @var array<array{day: int, time_range: string}> */
-    public array $days = [];
-
-    /** @var int[] */
-    public array $templates = [];
-
-    /** @var array<array{day_range: string, time_range: string}> */
-    public array $exceptions = [];
+    /**
+     * @param mixed $name
+     * @param mixed $alias
+     * @param array<array{day:int, time_range:string}>|null $days
+     * @param array<int>|null $templates
+     * @param array<array{day_range:string, time_range:string}>|null $exceptions
+     */
+    public function __construct(
+        #[Assert\NotNull]
+        #[Assert\Type('string')]
+        #[Assert\Length(min: 1, max: 200)]
+        public readonly mixed $name,
+        #[Assert\NotNull]
+        #[Assert\Type('string')]
+        #[Assert\Length(min: 1, max: 200)]
+        public readonly mixed $alias,
+        #[Assert\NotNull]
+        #[Assert\Type('array')]
+        #[Assert\All([
+            new Assert\Collection(
+                fields: [
+                    'day' => [
+                        new Assert\NotNull(),
+                        new Assert\Type('integer'),
+                    ],
+                    'time_range' => [
+                        new Assert\NotNull(),
+                        new Assert\Type('string'),
+                    ],
+                ],
+            ),
+        ])]
+        public readonly ?array $days,
+        #[Assert\NotNull]
+        #[Assert\Type('array')]
+        #[Assert\All(
+            new Assert\Type('integer'),
+        )]
+        public readonly ?array $templates,
+        #[Assert\NotNull]
+        #[Assert\Type('array')]
+        #[Assert\All([
+            new Assert\Collection(
+                fields: [
+                    'day_range' => [
+                        new Assert\NotNull(),
+                        new Assert\Type('string'),
+                    ],
+                    'time_range' => [
+                        new Assert\NotNull(),
+                        new Assert\Type('string'),
+                    ],
+                ],
+            ),
+        ])]
+        public readonly ?array $exceptions = [],
```

Comment:
Those 3 array properties can also be typed as mixed if you want the type validation to occur. 
Otherwise, a user input with string instead of the array will trigger a TypeError instead of the validation.